registry/remote/referrers.go

```diff
@@ -68,6 +70,25 @@ var (
 	errNoReferrerUpdate = errors.New("no referrer update")
 )
 
+// DanglingReferrerIndexError is returned when failed to delete old referrer index
+// after newly updated referrer index being pushed.
+// Only returned if referrer API is unavailable.
+type DanglingReferrerIndexError struct {
```

Comment:
Is this type really necessary and can we  return error codes? https://github.com/docker/distribution/blob/master/docs/spec/api.md#errors-2

Reply:
IMHO: The error code should be part for the response, oras-go should not modify it.

Also, we are trying to handle a situation goes beyond the distribution spec: how to do garbage collection for [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#referrers-tag-schema). The error code is defined by distribution-spec and we should not use it to handle a situation not covered by the spec.